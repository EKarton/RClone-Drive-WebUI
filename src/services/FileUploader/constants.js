export const ActionTypes = Object.freeze({
  UPLOAD_FILE: 'UPLOAD_FILE',
  CANCEL_UPLOAD: 'CANCEL_UPLOAD',
});
