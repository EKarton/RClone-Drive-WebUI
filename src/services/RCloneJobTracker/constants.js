export const ActionTypes = Object.freeze({
  TRACK_NEW_JOB: 'TRACK_NEW_JOB',
  UNTRACK_NEW_JOB: 'UNTRACK_NEW_JOB',
});
