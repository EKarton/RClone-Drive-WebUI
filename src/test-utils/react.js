import { FileViewerProvider } from 'contexts/FileViewerStore';
import { RCloneInfoStateProvider } from 'contexts/RCloneInfoStore';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router';

const customRender = (
  component,
  { initialFileViewerState, initialRCloneInfoState } = {},
  { route = '/' } = {}
) => {
  const history = createMemoryHistory({ initialEntries: [route] });

  const renderedComponent = render(
    <RCloneInfoStateProvider defaultState={initialRCloneInfoState}>
      <FileViewerProvider defaultState={initialFileViewerState}>
        <Router history={history}>{component}</Router>
      </FileViewerProvider>
    </RCloneInfoStateProvider>
  );

  return {
    ...renderedComponent,
    history,
  };
};

export * from '@testing-library/react';
export * from '@testing-library/user-event';
export { customRender, userEvent };
