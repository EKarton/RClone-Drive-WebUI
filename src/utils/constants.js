export const ICON_SIZE = Object.freeze({
  SMALL: 'small',
  MEDIUM: 'medium',
  LARGE: 'large',
});

export const StatusTypes = Object.freeze({
  LOADING: 'loading',
  SUCCESS: 'success',
  ERROR: 'error',
});

export const JobStatus = Object.freeze({
  ONGOING: 'ONGOING',
  SUCCESS: 'SUCCESS',
  ERROR: 'ERROR',
});

export const JobTypes = Object.freeze({
  UPLOAD_FILE: 'UPLOAD_FILE',
  MOVE_FILE: 'MOVE_FILE',
  MOVE_FOLDER: 'MOVE_FOLDER',
  RENAME_FILE: 'RENAME_FILE',
  RENAME_FOLDER: 'RENAME_FOLDER',
});

export const COLOR_MODE = Object.freeze({
  LIGHT: 'light',
  DARK: 'dark',
});

export const ImageMimeTypes = new Set(['image/jpeg', 'image/png']);
