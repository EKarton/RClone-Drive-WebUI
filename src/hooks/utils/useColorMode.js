import { useContext } from 'react';
import { ColorModeContext } from 'contexts/ColorMode/index';

export default function useColorMode() {
  const context = useContext(ColorModeContext);

  if (context === undefined) {
    throw new Error('useColorMode() needs to be used inside ColorModeProvider');
  }

  return context;
}
