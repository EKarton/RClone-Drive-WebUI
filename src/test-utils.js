import { FileViewerProvider, initialState } from 'contexts/FileViewerStore';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router';

const customRender = (
  component,
  { initialFileViewerState = initialState } = {},
  { route = '/' } = {}
) => {
  const history = createMemoryHistory({ initialEntries: [route] });

  return render(
    <FileViewerProvider defaultState={initialFileViewerState}>
      <Router history={history}>{component}</Router>
    </FileViewerProvider>
  );
};

export * from '@testing-library/react';
export * from '@testing-library/user-event';
export { customRender, userEvent };
