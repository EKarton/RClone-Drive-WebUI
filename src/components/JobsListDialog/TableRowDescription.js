import { JobStatus } from 'services/RCloneJobTracker/constants';
import { getFullPath } from 'utils/filename-utils';

const getFileString = (file) => {
  return getFullRemotePath(file.remote, getFullPath(file.dirPath, file.name));
};

const getFullRemotePath = (remote, fullPath) => {
  return `${remote}:${fullPath}`;
};

function Upload({ status, job }) {
  if (status === JobStatus.ERROR) {
    return (
      <span>
        Failed to upload file {job.name} to {job.remote}:{job.dirPath}
      </span>
    );
  }

  if (status === JobStatus.SUCCESS) {
    return (
      <span>
        Uploaded file {job.name} to {job.remote}:{job.dirPath}
      </span>
    );
  }

  return (
    <span>
      Uploading file {job.name} to {job.remote}:{job.dirPath}
    </span>
  );
}

function MoveFile({ status, job }) {
  if (status === JobStatus.ERROR) {
    return (
      <span>
        Failed to move file from {getFileString(job.src)} to {getFileString(job.target)}
      </span>
    );
  }

  if (status === JobStatus.SUCCESS) {
    return (
      <span>
        Moved file from {getFileString(job.src)} to {getFileString(job.target)}
      </span>
    );
  }

  return (
    <span>
      Moving file from {getFileString(job.src)} to {getFileString(job.target)}
    </span>
  );
}

function MoveFolder({ status, job }) {
  if (status === JobStatus.ERROR) {
    return (
      <span>
        Failed to move folder from {getFileString(job.src)} to {getFileString(job.target)}
      </span>
    );
  }

  if (status === JobStatus.SUCCESS) {
    return (
      <span>
        Moved folder from {getFileString(job.src)} to {getFileString(job.target)}
      </span>
    );
  }

  return (
    <span>
      Moving folder from {getFileString(job.src)} to {getFileString(job.target)}
    </span>
  );
}

function RenameFile({ status, job }) {
  if (status === JobStatus.ERROR) {
    return (
      <span>
        Failed to rename file {job.src.name} to {job.target.name} in {job.src.remote}:
        {job.src.dirPath}
      </span>
    );
  }

  if (status === JobStatus.SUCCESS) {
    return (
      <span>
        Renamed file {job.src.name} to {job.target.name} in {job.src.remote}:
        {job.src.dirPath}
      </span>
    );
  }

  return (
    <span>
      Renaming file {job.src.name} to {job.target.name} in {job.src.remote}:
      {job.src.dirPath}
    </span>
  );
}

function RenameFolder({ status, job }) {
  if (status === JobStatus.ERROR) {
    return (
      <span>
        Failed to rename folder {job.src.name} to {job.target.name} in {job.src.remote}:
        {job.src.dirPath}
      </span>
    );
  }

  if (status === JobStatus.SUCCESS) {
    return (
      <span>
        Renamed folder {job.src.name} to {job.target.name} in {job.src.remote}:
        {job.src.dirPath}
      </span>
    );
  }

  return (
    <span>
      Renaming folder {job.src.name} to {job.target.name} in {job.src.remote}:
      {job.src.dirPath}
    </span>
  );
}

function GenericJob({ status }) {
  if (status === JobStatus.ERROR) {
    return <span>Job failed</span>;
  }

  if (status === JobStatus.SUCCESS) {
    return <span>Job succeeded</span>;
  }

  return <span>Job ongoing</span>;
}

export default function TableRowDescription({ status, job }) {
  if (job.jobType === 'UPLOAD_FILE') {
    return <Upload status={status} job={job} />;
  }

  if (job.jobType === 'MOVE_FILE') {
    return <MoveFile status={status} job={job} />;
  }

  if (job.jobType === 'MOVE_FOLDER') {
    return <MoveFolder status={status} job={job} />;
  }

  if (job.jobType === 'RENAME_FILE') {
    return <RenameFile status={status} job={job} />;
  }

  if (job.jobType === 'RENAME_FOLDER') {
    return <RenameFolder status={status} job={job} />;
  }

  return <GenericJob status={status} />;
}
