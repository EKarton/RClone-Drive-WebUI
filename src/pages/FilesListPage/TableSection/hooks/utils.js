import { JobStatus, JobTypes } from 'utils/constants';

export function getFileNames(data) {
  return new Set(data.filter((file) => !file.IsDir).map((file) => file.Name));
}

export function getFolderNames(data) {
  return new Set(data.filter((file) => file.IsDir).map((file) => file.Name));
}

export function getUploadingJobs(jobs, remote, path) {
  return jobs.filter(
    (job) =>
      job.jobType === JobTypes.UPLOAD_FILE &&
      job.remote === remote &&
      job.dirPath.startsWith(path) &&
      job.status.value !== JobStatus.SUCCESS
  );
}

export function getUploadingFiles(jobs, remote, path) {
  return getUploadingJobs(jobs, remote, path).filter((job) => job.dirPath === path);
}

export function getUploadingFolders(jobs, remote, path) {
  return getUploadingJobs(jobs, remote, path).filter((job) => job.dirPath !== path);
}

export function getMovingJobs(jobs, remote, path) {
  const supportedJobTypes = new Set([
    JobTypes.MOVE_FILE,
    JobTypes.MOVE_FOLDER,
    JobTypes.RENAME_FILE,
    JobTypes.RENAME_FOLDER,
  ]);

  return jobs.filter((job) => {
    const correctJob = supportedJobTypes.has(job.jobType);
    const inProgress = job.status.value === JobStatus.ONGOING;

    if (!correctJob || !inProgress) {
      return false;
    }

    const isMatchSrc = job.src?.remote === remote && job.src?.dirPath === path;
    const isMatchTarget = job.target?.remote === remote && job.target?.dirPath === path;

    return correctJob && inProgress && (isMatchSrc || isMatchTarget);
  });
}
