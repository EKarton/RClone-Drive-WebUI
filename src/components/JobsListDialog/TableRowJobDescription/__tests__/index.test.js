import { JobStatus, JobTypes } from 'utils/constants';
import { render } from 'test-utils/react';
import TableRowDescription from '../index';

describe('TableRowJobDescription', () => {
  it.each([
    JobTypes.MOVE_FILE,
    JobTypes.MOVE_FOLDER,
    JobTypes.RENAME_FILE,
    JobTypes.RENAME_FOLDER,
    JobTypes.UPLOAD_FILE,
    'Random job',
  ])('should contain expected string given job type is %s', (jobType) => {
    const { baseElement } = render(
      <TableRowDescription
        status={JobStatus.ONGOING}
        job={{
          jobType,
          src: {},
          target: {},
          remote: '',
          dirPath: '',
          name: '',
        }}
      />
    );

    expect(baseElement).toMatchSnapshot();
  });
});
