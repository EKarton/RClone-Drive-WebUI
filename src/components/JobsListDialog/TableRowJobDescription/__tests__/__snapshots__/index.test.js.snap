// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRowJobDescription should contain expected string given job type is MOVE_FILE 1`] = `
<body>
  <div>
    <span>
      Moving file from 
      undefined:undefined
       to 
      undefined:undefined
    </span>
  </div>
</body>
`;

exports[`TableRowJobDescription should contain expected string given job type is MOVE_FOLDER 1`] = `
<body>
  <div>
    <span>
      Moving folder from 
      undefined:undefined
       to 
      undefined:undefined
    </span>
  </div>
</body>
`;

exports[`TableRowJobDescription should contain expected string given job type is RENAME_FILE 1`] = `
<body>
  <div>
    <span>
      Renaming file 
       to 
       in 
      :
    </span>
  </div>
</body>
`;

exports[`TableRowJobDescription should contain expected string given job type is RENAME_FOLDER 1`] = `
<body>
  <div>
    <span>
      Renaming folder 
       to 
       in 
      :
    </span>
  </div>
</body>
`;

exports[`TableRowJobDescription should contain expected string given job type is Random job 1`] = `
<body>
  <div>
    <span>
      Job ongoing
    </span>
  </div>
</body>
`;

exports[`TableRowJobDescription should contain expected string given job type is UPLOAD_FILE 1`] = `
<body>
  <div>
    <span>
      Uploading file 
      
       to 
      
      :
      
    </span>
  </div>
</body>
`;
